specification/cognitiveservices/OpenAI.Inference/models/completions/chat_messages.tsp

```diff
@@ -0,0 +1,202 @@
+  @doc("Specifies that the model should determine which detail level to apply using heuristics like image size.")
+  auto: "auto",
+
+  @doc("""
+  Specifies that image evaluation should be constrained to the 'low-res' model that may be faster and consume fewer
+  tokens but may also be less accurate for highly detailed images.
+  """)
+  low: "low",
+
+  @doc("""
+  Specifies that image evaluation should enable the 'high-res' model that may be more accurate for highly detailed
+  images but may also be slower and consume more tokens.
+  """)
+  high: "high",
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A structured chat content item containing an image reference.")
+model ChatMessageImageContentItem extends ChatMessageContentItem {
+  @doc("The discriminated object type: always 'image_url' for this type.")
+  type: "image_url";
+
+  @doc("An internet location, which must be accessible to the model,from which the image may be retrieved.")
+  url: url;
+
+  @doc("""
+  The evaluation quality setting to use, which controls relative prioritization of speed, token consumption, and
+  accuracy.
+  """)
+  detail?: ChatMessageImageDetailLevel;
+}
+
+@discriminator("role")
+@added(ServiceApiVersions.v2023_05_15)
+@doc("An abstract representation of a chat message as provided in a request.")
+model ChatRequestMessage {
+  @doc("The chat role associated with this message.")
+  role: ChatRole;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("""
+A request chat message containing system instructions that influence how the model will generate a chat completions
+response.
+""")
+model ChatRequestSystemMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'system' for system messages.")
+  role: ChatRole.system;
+
+  @doc("The contents of the system message.")
+  content: string;
+
+  @doc("An optional name for the participant.")
+  name?: string;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A request chat message representing user input to the assistant.")
+model ChatRequestUserMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'user' for user messages.")
+  role: ChatRole.user;
+
+  @doc("The contents of the user message, with available input types varying by selected model.")
+  content: ChatMessageContent;
+
+  @doc("An optional name for the participant.")
+  name?: string;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A request chat message representing response or action from the assistant.")
+model ChatRequestAssistantMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'assistant' for assistant messages.")
+  role: ChatRole.assistant;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "explicitly nullable in mirrored API"
+  @doc("The content of the message.")
+  content: string | null;
+
+  @doc("An optional name for the participant.")
+  name?: string;
```

Comment:
Ok this is probably a non-Azure field. Sorry for all the noise. I will leave the comments just in case.

Reply:
Right, these all show up in [OpenAI's docs](https://platform.openai.com/docs/api-reference/chat/create#chat-create-messages), though they curiously don't seem to appear [in the matching spec](https://github.com/openai/openai-openapi/blob/a7a0b63d1fac3acac9a223d57da1219c8a97c01e/openapi.yaml#L5617C9-L5617C9). I know these for sure worked in the past (you could provide a name in the user message and then the assistant would "know your name") so we should likely leave these to begin with and follow up to see if removal was intended.